packages/web/src/pages/chat-page/components/ChatMessageTrack.tsx

```diff
@@ -0,0 +1,113 @@
+import { useCallback } from 'react'
+
+import {
+  Kind,
+  Status,
+  makeUid,
+  Track,
+  queueSelectors,
+  playerSelectors,
+  Name,
+  PlaybackSource,
+  queueActions,
+  QueueSource
+} from '@audius/common'
+import { useDispatch, useSelector } from 'react-redux'
+
+import { make } from 'common/store/analytics/actions'
+import MobileTrackTile from 'components/track/mobile/ConnectedTrackTile'
+
+const { makeGetCurrent } = queueSelectors
+const { getPlaying } = playerSelectors
+const { add, play, pause } = queueActions
+
+type ChatMessageTrackProps = {
+  track: Track | undefined | null
+  status: Status
+  errorMessage: string | null | undefined
+}
+
+export const ChatMessageTrack = ({
+  track,
+  status,
+  errorMessage
+}: ChatMessageTrackProps) => {
+  const dispatch = useDispatch()
+  const currentQueueItem = useSelector(makeGetCurrent())
+  const uid = track ? makeUid(Kind.TRACKS, track.track_id) : ''
+  const playing = useSelector(getPlaying)
+  const isTrackPlaying =
+    playing &&
+    !!track &&
+    !!currentQueueItem.track &&
+    currentQueueItem.track.track_id === track.track_id
+
+  const recordAnalytics = useCallback(
+    ({ name, source }: { name: Name; source: PlaybackSource }) => {
+      if (!track) return
+      dispatch(
+        make(name, {
+          id: `${track.track_id}`,
+          source
+        })
+      )
+    },
+    [dispatch, track]
+  )
+
+  const onTogglePlay = useCallback(() => {
+    if (!track) return
+    if (isTrackPlaying) {
+      dispatch(pause({}))
+      recordAnalytics({
+        name: Name.PLAYBACK_PAUSE,
+        source: PlaybackSource.DM_TRACK
+      })
+    } else if (
+      currentQueueItem.uid !== uid &&
+      currentQueueItem.track &&
+      currentQueueItem.track.track_id === track.track_id
+    ) {
+      dispatch(play({}))
+      recordAnalytics({
+        name: Name.PLAYBACK_PLAY,
+        source: PlaybackSource.DM_TRACK
+      })
+    } else {
+      dispatch(
+        add({
+          entries: [{ id: track.track_id, uid, source: QueueSource.DM_TRACKS }]
+        })
+      )
+      dispatch(play({ uid }))
+      recordAnalytics({
+        name: Name.PLAYBACK_PLAY,
+        source: PlaybackSource.DM_TRACK
+      })
+    }
+  }, [dispatch, recordAnalytics, track, isTrackPlaying, currentQueueItem, uid])
+
+  if (status === Status.ERROR) {
+    return (
+      // todo
+      <div>error</div>
+    )
+  }
+
+  return (
+    // You may wonder why we use the mobile web track tile here.
+    // It's simply because the DMs track tile uses the mobile web version.
```

Comment:
love this explainer haha

super minor nitpicky thing, maybe `uses the same design as mobile web`?